src/main/java/ma/hibernate/dao/PhoneDaoImpl.java

```diff
@@ -12,11 +20,59 @@ public PhoneDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Phone create(Phone phone) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(phone);
+            transaction.commit();
+        } catch (HibernateException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException(
+                    "Can't create phone " + phone, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return phone;
     }
 
     @Override
     public List<Phone> findAll(Map<String, String[]> params) {
-        return null;
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder cb = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> query = cb.createQuery(Phone.class);
+            Root<Phone> root = query.from(Phone.class);
+            List<Predicate> predicates = new LinkedList<>();
+            for (Map.Entry<String, String[]> param : params.entrySet()) {
+                CriteriaBuilder.In<String> paramsPredicate = cb.in(root.get(param.getKey()));
+                for (String valueParam : param.getValue()) {
+                    paramsPredicate.value(valueParam);
+                }
+                predicates.add(paramsPredicate);
+            }
+            query.where(cb.and(predicates.toArray(new Predicate[0])));
+            return session.createQuery(query).getResultList();
+        } catch (HibernateException e) {
+            String paramsString = mapToParams(params);
+            throw new RuntimeException(
+                    "Can't get phones from DB by parameters: "
+                            + paramsString, e);
+        }
+    }
+
+    private String mapToParams(Map<String, String[]> params) {
```

Comment:
it's redundant to create a method for this. You can use Stream API for the printing of this information